modepy/modes.py

```diff
@@ -656,6 +687,11 @@ def tensor_product_basis(dims, basis_1d):
 
     .. versionadded:: 2017.1
     """
+    warn("tensor_product_basis is deprecated. "
+            "Use TensorProductBasis instead. "
```

Comment:
```suggestion
            "Use basis_for_shape instead. "
```
? or make `TensorProductBasis` public?

Reply:
7f4e1a1 and 347027b